core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedResultCollector.java

```diff
@@ -148,8 +148,8 @@ public DictionaryBasedResultCollector(BlockExecutionInfo blockExecutionInfos) {
         // get the complex columns data of this row
         fillComplexColumnDataBufferForThisRow();
         for (int i = 0; i < queryDimensions.length; i++) {
-          fillDimensionData(scannedResult, surrogateResult, noDictionaryKeys, complexTypeKeyArray,
-              comlexDimensionInfoMap, row, i);
+          fillDimensionData(queryDimensions[i], scannedResult, surrogateResult,
```

Comment:
please handle the default values for the restructured blocks in RestructureBasedDictionaryResultCollector

Reply:
why it needs to handle the default values for the newly added complex columns? Currently CarbonData can't support to add complex columns for table, this pr just resolves the problem of  adding non-complex columns for a table which has complex columns.